### PR TITLE
implements the project list

### PR DESCRIPTION
**Pull request by janitha09**
This change implements project list endpoint

### CODE REVIEW

**Comment by noam-codefresh** [comment]
Hello @janitha09 - I apologize for letting this PR fall between the cracks. We would be happy to merge it into the master branch. Can you please merge/rebase it, and we will have a look again?

**Comment by janitha09** [comment]
@noam-codefresh merge conflicts and API changes have been resolved for

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "go.testFlags": ["-v"]
+}
```
Comment (noam-codefresh) [inline .vscode/settings.json:3]: please do not include this file in the pr
I see the repo doesn't have a .gitignore file - maybe even include one that ignores visual studio code and golang artifacts? (just a suggestion)

File: pkg/codefresh/project_test.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	Auth: AuthOptions{
+		Token: "",
+	},
+	Debug: true,
+	Host:  "https://g.codefresh.io",
+}
+
+func TestProject(t *testing.T) {
+
+	cf := New(&clientOptions)
```
Comment (noam-codefresh) [inline pkg/codefresh/project_test.go:19]: maybe you can use the codefresh/mock/codefresh struct to make this test better?
you will also need to regenerate the mock itself to include your `Projects()` func. You will need mockery 1.1.1 installed, and run `go generate ./...` for it.
let me know if you need any help with it.

Comment (janitha09) [reply pkg/codefresh/project_test.go:19]: @noam-codefresh is there an example that shows the coddfresh mock structs use.

For an API client that should have been code generated, this abstraction is proving to be rather tedious, and hard to follow, it most certainly confuses the compiler
